src/library.js

```diff
@@ -2388,6 +2388,23 @@ addToLibrary({
   _emscripten_get_now_is_monotonic__deps: ['$nowIsMonotonic'],
   _emscripten_get_now_is_monotonic: () => nowIsMonotonic,
 
+  __emscripten_atomics_sleep__internal: true,
+  // In a browser without cross origin isolation, SharedArrayBuffer is deleted
+  // from the global scope:
+  // https://html.spec.whatwg.org/multipage/webappapis.html#creating-a-new-javascript-realm
+  // However, it says:
+  //
+  // > Web developers can still get at the constructor through
+  // > `new WebAssembly.Memory({ shared:true, initial:0, maximum:0}).buffer.constructor`.
+  //
+  // That is what we are up to here. Use an IIFE to avoid shadowing
+  // SharedArrayBuffer globally.
+  __emscripten_atomics_sleep__postset: `
+    var SABConstructor = new WebAssembly.Memory({"shared":true,"initial":0,"maximum":0}).buffer.constructor;
```

Comment:
Is the new keyword needed here?  Does does it mean to create a "new" constructor function?

Reply:
Yeah it's needed: we create a new empty wasm shared memory and then look up the constructor of the backing buffer of it. So if you inserted parenthesis they would go like 
```js
(new WebAssembly.Memory({"shared":true,"initial":0,"maximum":0})).buffer.constructor
```
The syntactic behavior of `new` is a bit odd -- it seems to modify the first function call in the expression or something.